src/app.js

```diff
@@ -1,21 +1,59 @@
-'use strict';
-
-/**
- * Implement sum function:
- *
- * Function takes 2 numbers and returns their sum
- *
- * sum(1, 2) === 3
- * sum(1, 11) === 12
- *
- * @param {number} a
- * @param {number} b
- *
- * @return {number}
- */
-function sum(a, b) {
-  // write code here
-  return a + b;
-}
-
-module.exports = sum;
+'use strict'
+
+const http = require('http');
+const fs = require('fs');
+const path = require('path');
+
+const PORT = process.env.port || 8080;
+
+const BASE_FILE_PATH = '../public';
+
+const WRONG_PATH_ERROR = 'Wrong path to the file. '
+ + 'There is no file in the public folder with given path.';
+const PATHNAME_ERROR = 'To get the file URL path should start with \'/file\'';
+const UNKNOWN_ERROR = 'Unknown error. Error message:';
+
+const server = http.createServer((req, res) => {
+  const { pathname } = new URL(req.url, `http://${req.headers.host}`);
+
+  const errors = [];
+  const response = {};
+  
+  const sendSerializedErrorResponse = () => {
+    req.statusCode = 404;
+    res.statusMessage = 'Not Found';
+    res.setHeader('Content-Type', 'application/json');
+
+    response.errors = errors;
+
+    res.end(JSON.stringify(response));
+  };
+
+  const isPathnameValid = pathname.startsWith('/file');
+
+  if (!isPathnameValid) {
+    errors.push({ message: PATHNAME_ERROR });
+    sendSerializedErrorResponse();
+    return;
+  }
+
+  const filePath = pathname.slice(6);
```

Comment:
to remove `'/file/'` - with slash in the end
otherwise, I have to add `'.'` at the beginning to get correctly resolved path

`./index.html`, `index.html` resolve correct
`/index.html` resolves incorrect